### ISSUE TITLE
Strange nesting when discussion has nested headings

### ISSUE THREAD

**Issue by yuvipanda** [kjschiroo/WikiChatter#12]
https://en.wikipedia.org/wiki/Wikipedia:Teahouse/Questions/Archive_143#Deletionism.3F produces:

```
== Deletionism? ==
=== Deletionism? (Edit Break 1) ===
    User:AugurNZ | Time:01:51, 18 September 2013 (UTC)
        User:Writ_Keeper | Time:02:16, 18 September 2013 (UTC)
            User:AugurNZ | Time:02:52, 18 September 2013 (UTC)
                User:Cullen328 | Time:03:02, 18 September 2013 (UTC)
                    User:Cullen328 | Time:03:29, 18 September 2013 (UTC)
                        User:AugurNZ | Time:03:46, 18 September 2013 (UTC)
                            User:LtPowers | Time:14:38, 18 September 2013 (UTC)
                                User:Safetyprofessional | Time:19:30, 18 September 2013 (UTC)
    User:AugurNZ | Time:20:57, 18 September 2013 (UTC)
        User:Anne Delong | Time:21:03, 18 September 2013 (UTC)
            User:AugurNZ | Time:21:52, 18 September 2013 (UTC)
                User:Konveyor_Belt | Time:22:45, 18 September 2013 (UTC)
                    User:AugurNZ | Time:22:55, 18 September 2013 (UTC)
                        User:LukeSurl | Time:23:08, 18 September 2013 (UTC)
                            User:LukeSurl | Time:23:20, 18 September 2013 (UTC)
                                User:Konveyor_Belt | Time:23:27, 18 September 2013 (UTC)
    User:AugurNZ | Time:23:37, 18 September 2013 (UTC)
        User:Writ_Keeper | Time:23:52, 18 September 2013 (UTC)
            User:AugurNZ | Time:00:17, 19 September 2013 (UTC)
                User:Writ_Keeper | Time:00:53, 19 September 2013 (UTC)
                    User:AugurNZ | Time:01:30, 19 September 2013 (UTC)
                        User:Salvidrim | Time:02:04, 19 September 2013 (UTC)
                            User:AugurNZ | Time:18:34, 19 September 2013 (UTC)
```

where I'd expect the secondary heading to be somewhere in the middle. Not sure if this is a parsing issue or a pretty-print issue

**Comment by yuvipanda** [comment]
Also in https://en.wikipedia.org/wiki/Wikipedia:Teahouse/Questions/Archive_420#Draft:Mike_.22Greeny.22_Green_.28again.29 so consistently reproducible.

**Comment by yuvipanda** [comment]
This and https://github.com/earwig/mwparserfromhell/issues/55 are the only obvious issues I still have left from all the Teahouse archives. Given how fickle these are, amazing work :D

**Comment by yuvipanda** [comment]
Where was this fixed? I'm trying to find the commit...

**Comment by kjschiroo** [comment]
It looks like I didn't quiet give this enough attention when I was first looking at it. I had thought that it would also be fixed by pull request #11. After merging that I am now getting this output

```
== Deletionism? ==
=== Deletionism? (Edit Break 1) ===
    User:AugurNZ | Time:01:51, 18 September 2013 (UTC)
        User:Writ_Keeper | Time:02:16, 18 September 2013 (UTC)
            User:AugurNZ | Time:02:52, 18 September 2013 (UTC)
                User:Cullen328 | Time:03:02, 18 September 2013 (UTC)
                    User:Cullen328 | Time:03:29, 18 September 2013 (UTC)
                        User:AugurNZ | Time:03:46, 18 September 2013 (UTC)
                            User:LtPowers | Time:14:38, 18 September 2013 (UTC)
                                User:Safetyprofessional | Time:19:30, 18 September 2013 (UTC)
    User:AugurNZ | Time:20:57, 18 September 2013 (UTC)
        User:Anne Delong | Time:21:03, 18 September 2013 (UTC)
            User:AugurNZ | Time:21:52, 18 September 2013 (UTC)
                User:Konveyor_Belt | Time:22:45, 18 September 2013 (UTC)
                    User:AugurNZ | Time:22:55, 18 September 2013 (UTC)
                        User:LukeSurl | Time:23:08, 18 September 2013 (UTC)
                            User:LukeSurl | Time:23:20, 18 September 2013 (UTC)
                                User:Konveyor_Belt | Time:23:27, 18 September 2013 (UTC)
    User:AugurNZ | Time:23:37, 18 September 2013 (UTC)
        User:Writ_Keeper | Time:23:52, 18 September 2013 (UTC)
            User:AugurNZ | Time:00:17, 19 September 2013 (UTC)
                User:Writ_Keeper | Time:00:53, 19 September 2013 (UTC)
                    User:AugurNZ | Time:01:30, 19 September 2013 (UTC)
                        User:Salvidrim | Time:02:04, 19 September 2013 (UTC)
                            User:AugurNZ | Time:18:34, 19 September 2013 (UTC)
=== Deletionism? (Edit Break 1) ===
    User:AugurNZ | Time:23:37, 18 September 2013 (UTC)
        User:Writ_Keeper | Time:23:52, 18 September 2013 (UTC)
            User:AugurNZ | Time:00:17, 19 September 2013 (UTC)
                User:Writ_Keeper | Time:00:53, 19 September 2013 (UTC)
                    User:AugurNZ | Time:01:30, 19 September 2013 (UTC)
                        User:Salvidrim | Time:02:04, 19 September 2013 (UTC)
                            User:AugurNZ | Time:18:34, 19 September 2013 (UTC)
```

So `=== Deletionism? (Edit Break 1) ===` ends up in its super-section as well as just underneath it. I think that there is an argument in `mwparserfromhell.get_sections` that should let me fix it pretty easily.

Definitely my bad, I shouldn't have closed this one yet. Reopening. 

This input also illustrates a problem with `{{od|:::::::}}`, but I will open a separate issue for that.

**Comment by kjschiroo** [comment]
Commit message was a little confusing, real issue is fixed in https://github.com/kjschiroo/WikiChatter/commit/08c2e38f9c87926476e7ad346001bf2a8271ab47
